js/src/admin/components/ExtensionSettingsPage.tsx

```diff
@@ -108,7 +138,108 @@ export default class TrelloSettingsPage extends ExtensionPage {
     m.redraw();
   }
 
+  getTag(id: string): Tag | undefined {
+    return this.states.tags?.find((tag) => tag.id() === String(id));
+  }
+
+  addNewTagMappingButton(): JSX.Element {
+    return (
+      <Button
+        disabled={this.loading}
+        class="Button"
+        onclick={() =>
+          app.modal.show(NewTagMappingModal, {
+            states: this.states,
+            onClose: (boardShortLink: string, label: {}, tagId: string) => {
+              if (this.states.mappings[boardShortLink] === undefined) {
+                this.states.mappings[boardShortLink] = [];
+              }
+
+              const exists = this.states.mappings[boardShortLink].filter((value) => value.tagId == tagId || value.label.id == label.id);
+
+              if (!exists.length) {
+                this.states.mappings[boardShortLink].push({ label, tagId });
+
+                this.states.dirty = true;
+                m.redraw();
+              }
+            },
+          })
+        }
+      >
+        {app.translator.trans('blomstra-trello.admin.settings.mapping.add_new')}
+      </Button>
+    );
+  }
+
+  saveButton(): JSX.Element {
```

Comment:
```suggestion
  saveButton(): Mithril.Children {
```

Reply:
@davwheat can you explain why we are using this instead of JSX.Element? I'm asking since in the previous project as an example it's using JSX.Element.